### PR TITLE
Fix for PR 5754

### PR DESCRIPTION
**Pull request by goplayoutside3**
See context for these changes in PR #5754's discussion, and particularly my [comment here](https://github.com/zooniverse/front-end-monorepo/pull/5754#issuecomment-1944310442).

### CODE REVIEW

**Comment by kieftrav** [comment]
@goplayoutside3 - just tested out your PR and the bug I outlined is still there. Please see the attached video about how previous marks on different frames don't get pulled in the way I assume they should.

https://github.com/zooniverse/front-end-monorepo/assets/733986/7f37d167-1e47-4305-ac8e-4a8027ebdf1b

**Comment by goplayoutside3** [comment]
Ah okay I see what's happening. The video is related to your question about `previousInteractionTaskAnnotations()` [here](https://github.com/zooniverse/front-end-monorepo/pull/5754#issuecomment-1944310442). In the subject viewer, there are two different ways `marks` are grabbed from the store:

1) InteractionLayerContainer

InteractionLayerContainer gets its `marks` from `classifierStore.workflowSteps.active.tasks`. "active" is a Step type in this case, and `tasks` is an array of type DrawingTask. `get marks()` is a "view" on the DrawingTask model that loops over `self.tools` and returns all `tool.marks`. For example, this is relevant for a drawing task that requires a volunteer to draw one circle _and_ one point. `tool.marks` gets updated when `createMark()` from the Tool type is called in a UI component. Therefore, when a volunteer clicks on the InteractionLayer, that interaction calls `activeTool.createMark()` with all the necessary data to add a new `mark` to `tool.marks` in the store. 

Because InteractionLayerContainer is observing the store, it re-renders whenever `tool.marks` updates. Also, its child component InteractionLayer handles filtering per `frame` and renders the `marks` in DrawingToolMarks.

2) PreviousMarks

PreviousMarks gets its `marks` by calling `previousInteractionTaskAnnotations()` which grabs annotations from `classifierStore.classification.annotations`. That function also filters for `annotations` that do not match the active tasks's key (e.g `T1`). PreviousMarks observes the store too, so it should re-render whenever `previousAnnotations` updates, and handle filtering of those annotation marks by `frame`. There's a DrawingToolMarks component rendered in PreviousMarks for every annotation intentionally passing `disabled` as true.

You can inspect `annotations` by calling `self.toSnapshot()` inside `previousInteractionTaskAnnotations()`. While debugging, I found that `annotations` in PreviousMarks did contain expected data when going back and forth between workflow steps. So that narrows down the bug to PreviousMarks. I think that this line:
https://github.com/zooniverse/front-end-monorepo/blob/6b421edae1e92322bf3676cc558612e886983321/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js#L49
does not actually have access to `frame` on each Mark without more explicit handling of store snapshots. See the use of `resolveIdentifier` in DrawingAnnotation:
https://github.com/zooniverse/front-end-monorepo/blob/6b421edae1e92322bf3676cc558612e886983321/packages/lib-classifier/src/plugins/tasks/drawing/models/DrawingAnnotation.js#L17
and my logging of the `annotation` snapshot here:


To summarize, the reason InteractionLayerContainer and PreviousMarks grab annotation marks from the store in two different ways is because PreviousMarks is limited to only rendering marks created or edited by a volunteer and also needs to grab annotations from other steps. InteractionLayerContainer only cares about the active step.

I can't spend more time on this today, but will update if I think of a way to address this problem later this week.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js [hunk 1]
```diff
@@ -38,26 +35,25 @@ function PreviousMarks ({
...
-    // React v16 allows this, though. 
     return (
       <>
         {previousAnnotations.map((annotation) => {
-          const annotationValuesPerFrame = (multiImageCloneMarkers)
-            ? annotation.value
-            : annotation.value.filter(value => value.frame === frame)
-          
+          const annotationMarks = annotation.toSnapshot()[0].value
+          console.log(annotationMarks)
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js:52]: This shows correctly stored `marks` while jumping between workflow steps. The issue is that PreviousMarks isn't actually filtering per `frame` yet.

**Comment by goplayoutside3** [comment]
@kieftrav noting here that the bug in your video exists on production too. It can be replicated in the [How Did We Get Here](https://www.zooniverse.org/projects/communitiesandcrowds/how-did-we-get-here?demo=true) project that has multi-frame transcription subjects and two steps in the workflow.

**Comment by goplayoutside3** [comment]
I've asked Jim for input [here](https://github.com/zooniverse/front-end-monorepo/pull/2001/files#r1490354564), too.

**Comment by kieftrav** [comment]
@goplayoutside3 - I've incorporated Jim's one-liner into my PR. I don't think all of the other changes here make sense - they're focused on changing the `frame` on the container element instead of focusing on the `mark` itself having `frame=0`. Because modifying the frame is a data export issue decided in [Issue #5493](https://github.com/zooniverse/front-end-monorepo/issues/5493) and is confined to the mark itself, I think it's important to keep the comment about the GitHub issue next to the area that sets its value to 0 instead of creating a chain of assumptions about why we changed the frame itself to 0 for the whole `InteractionLayer` when all that matters is that the classification export registers the mark's frame as 0 upon creation. I think this is particularly poignant because allowing the frame number to be set to its real frame when `multiImageCloneMarkers` is true doesn't change behavior at all - it still works as expected cloning all marks across all frames.

Because we resolved the issue with losing track of marks in the PreviousMarks component (i.e. Jim's fix) the current PR works as expected as far as I can tell.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js:52]: Here, you should use [`getSnapshot(annotation)`](https://mobx-state-tree.js.org/concepts/snapshots) to debug the annotation. That's how I figured out that the annotation value didn't include all of the drawn marks.

`annotation.toSnapshot()` serialises a drawing annotation so that it can be saved to Panoptes.
